src/include/ipc4/notification.h

```diff
@@ -42,6 +42,17 @@ enum sof_ipc4_notification_type {
 	SOF_IPC4_MANAGEMENT_SERVICE		= 16,
 };
 
+enum sof_ipc4_resource_event_type {
+	SOF_IPC4_GATEWAY_UNDERRUN_DETECTED	= 6,
+	SOF_IPC4_GATEWAY_OVERRUN_DETECTED	= 7,
+};
+
+/* Resource Type - source of the event */
+enum sof_ipc4_resource_type {
+	/* Gateway */
+	GATEWAY = 2,
+};
+
```

Comment:
I guess these are generic definitions and could be a separate patch that adds the gateway resource event types to the common header. And then a second chain-dma specific patch that uses these flags for chain-dma.

Reply:
@kv2019i Done in #7023